lib/c0/c0sk_internal.c

```diff
@@ -372,6 +342,49 @@ c0sk_cningest_cb(void *rock, struct bonsai_kv *bkv, struct bonsai_val *vlist)
     if (ev(err))
         return err;
 
+    ingest->c0iw_kbytes += klen;
+    ingest->c0iw_vbytes += vlen;
+
+    if (ingest->c0iw_kbytes + ingest->c0iw_vbytes > ingest->c0iw_mask) {
+        struct throttle_tls *tls = ingest->c0iw_thr_tls;
+
+        tls->bytes += klen + vlen;
+
+        if (tls->bytes > ingest->c0iw_mask) {
+            struct throttle_sensor *ts = ingest->c0iw_thr_sensor;
+            uint64_t resid = tls->bytes & ingest->c0iw_mask;
+            uint64_t gen;
+
+            tls->bytes -= resid;
+
+            gen = atomic_read_acq(ts->ts_cntrgenp);
+            if (gen > tls->cntrgen) {
+                atomic_ulong *cntrp = &ts->ts_cntrv[gen % NELEM(ts->ts_cntrv)];
```

Comment:
Maybe move the assignment out of line since pct's is?

Reply:
i moved it for you, hope it doesn't break anything...